care/facility/api/serializers/facility.py

```diff
@@ -104,3 +109,27 @@ class Meta:
     def create(self, validated_data):
         validated_data["created_by"] = self.context["request"].user
         return super().create(validated_data)
+
+
+class FacilityImageUploadSerializer(serializers.ModelSerializer):
+    cover_image = serializers.ImageField(required=True, write_only=True)
+
+    class Meta:
+        model = Facility
+        fields = ("cover_image",)
+
+    def save(self, **kwargs):
+        facility = self.instance
+        image = self.validated_data["cover_image"]
+        s3 = boto3.client(
+            "s3",
+            **cs_provider.get_client_config(cs_provider.BucketType.FACILITY.value)
+        )
+        upload_response = s3.put_object(
+            Bucket=settings.FACILITY_S3_BUCKET,
+            Key=f"cover_images/{image.name}",
+            Body=image.file,
+        )
+        facility.cover_image_url = f"{upload_response['ResponseMetadata']['HTTPHeaders']['location']}cover_images/{image.name}"
```

Comment:
Ideally we want to store the key and store the CDN URL separately, we can ignore this for this instance i think.